File: backend/app/schemas/attachment.py
from datetime import datetime
from typing import Annotated, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, StringConstraints

AttachmentEntityType = Literal[
    "company",
    "contact",
    "lead",
    "deal",
    "activity",
    "task",
    "note",
]
AttachmentFilename = Annotated[str, StringConstraints(min_length=1, max_length=255)]
AttachmentContentType = Annotated[str, StringConstraints(min_length=1, max_length=255)]


class AttachmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    organization_id: UUID
    uploaded_by_user_id: UUID
    entity_type: AttachmentEntityType
    entity_id: UUID
    original_filename: AttachmentFilename
    content_type: AttachmentContentType
    size_bytes: int
    created_at: datetime

File: frontend/src/types/attachment.ts
import type { PaginationParams } from "@/types/pagination";

export const attachmentEntityTypes = [
  "company",
  "contact",
  "lead",
  "deal",
  "activity",
  "task",
  "note",
] as const;

export type AttachmentEntityType = (typeof attachmentEntityTypes)[number];

export type Attachment = {
  content_type: string;
  created_at: string;
  entity_id: string;
  entity_type: AttachmentEntityType;
  id: string;
  organization_id: string;
  original_filename: string;
  size_bytes: number;
  uploaded_by_user_id: string;
};

export type AttachmentListParams = PaginationParams & {
  entity_id: string;
  entity_type: AttachmentEntityType;
};
